File: WittyShare/WittyShare/Models/Comment.cs
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WittyShare.Models
{
    [BsonIgnoreExtraElements]
    public class Comment
    {
        public Comment(string commentData, Person commentedBy, int upVote, int downVote, bool verified)
        {
            CommentData = commentData;
            CommentedBy = commentedBy;
            UpVote = upVote;
            DownVote = downVote;
            Verified = verified;
        }

        public string CommentData { get; set; }
        public Person CommentedBy { get; set; }
        public int UpVote { get; set; }
        public int DownVote { get; set; }
        public Boolean Verified { get; set; }

    }
}

File: WittyShare/WittyShare/ClientApp/app/components/models/comment.ts
import { Person } from './person';
export class Comment {
    commentData: string;
    commentedBy: Person;
    upVote: number;
    downVote: number;
    verified: boolean;
    password: string;
    profession: string;
    experience: number;
    degree: string;
    degreeMajor: string;
    dob: string;
    

    constructor(data: any) {
        if (data) {
            data = data || {};
            this.commentData = data.commentData;
            this.commentedBy = new Person(data.commentedBy);
            this.upVote = data.upVote;
            this.downVote = data.downVote;
            this.verified = data.verified;
       
        }
    }
}     
